iris_grib/message.py

```diff
@@ -33,9 +33,24 @@
 from iris._lazy_data import as_lazy_data
 from iris.exceptions import TranslationError
 
+
 _SUPPORTED_GRID_DEFINITIONS = (0, 1, 5, 10, 12, 20, 30, 40, 90)
 
 
+def grib_get_array_wrapper(f):
+    """
+    Converts what f returns into an array.
+
+    When using eccodes-python, gribapi.grib_get_*array returns a list rather
+    than an array as expected.
+
+    This is intended as a *temporary* fix that wil be moved once eccodes-python
+    is fixed.
+
+    """
+    return np.array(f)
```

Comment:
@lbdreyer There a danger here that by only casting to `np.array` regardlessly we could lose a masked array that's passed in.

Should be be more selective?

Reply:
You're right. We could either check its an array before casting, or perhaps use `np.asanyarray` which allows masked arrays to keep the mask:
```
>>> a  = ma.array([1,2,3], mask=[0,1,1])
>>> a
masked_array(data = [1 -- --],
             mask = [False  True  True],
       fill_value = 999999)

>>> f = np.asanyarray(a)
>>> f
masked_array(data = [1 -- --],
             mask = [False  True  True],
       fill_value = 999999)
```